pandas/tests/frame/test_timeseries.py

```diff
@@ -706,6 +727,40 @@ def test_between_time_raises(self):
         with pytest.raises(TypeError):  # index is not a DatetimeIndex
             df.between_time(start_time='00:00', end_time='12:00')
 
+    @pytest.mark.parametrize('time_axis', [
```

Comment:
same this is very hard to read

Reply:
you can make the error case a separate test, which makes both tests much simpler